### PR TITLE
642 Document explaining dApps

### PR DESCRIPTION
**Pull request by ACStoneCL**
### Related links

[Create a documentation explaining what a dApp is #642](https://github.com/casper-network/docs/issues/642)

### Changes

Added a new document that explains what a decentralized application is, the difference between centralized and decentralized and basics of interaction with a Casper network.

Also spotted a typo in the SDK and fixed it.

### Notes

Ran locally without issue.

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/building-dapps/dapp.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# What is a dApp?
+
+A decentralized application (dApp) is an internet connected application with some data on a blockchain or similar peer-to-peer network.
+
+The degree that a dApp interacts with the blockchain can vary greatly depending on the needs of the application. Some dApps may use the blockchain simply to store data, with most of the computation and interaction taking place off-chain. Others may depend on logic stored on the blockchain, with only the bare minimum user interface stored outside of the blockchain itself.
+
+## Centralized Versus Decentralized
+
+A centralized system is one in which a single entity controls all aspects of the system. In some way, shape or form there is a single point of failure. Users are required to access this central entity to gain access to any involved application. The smallest subset of a centralized system would be a single machine running a single application. If the machine fails, the application will no longer function.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/building-dapps/dapp.md:9]:
```suggestion
A centralized system is one in which a single entity controls all aspects of the system. There is a single point of failure in some way, shape, or form. Users are required to access this central entity to gain access to any involved application. The smallest subset of a centralized system would be a single machine running a single application. If the machine fails, the application will no longer function.
```

File: source/docs/casper/dapp-dev-guide/building-dapps/dapp.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+A semi-decentralized system may delegate different sub-systems to a group of machines or layer of applications that specialize in specific tasks. However, these specialized machines still represent a potential single point of failure, where a component of the system may cease to function and result in the inability to use the overall system.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/building-dapps/dapp.md:11]:
```suggestion
A semi-decentralized system may delegate different sub-systems to a group of machines or layer of applications that specialize in specific tasks. However, these specialized machines still represent a potential single point of failure, where a system component may cease to function and result in the inability to use the overall system.
```

File: source/docs/casper/dapp-dev-guide/building-dapps/dapp.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+In contrast, a decentralized system consists of a group of [fungible](/glossary/F/#fungible) machines that can each perform as a full system or `distributed database`. Additional machines strengthen the overall system by adding additional redundancy and computational power.
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/building-dapps/dapp.md:13]: Can we simply say "interchangeable"?
```suggestion
In contrast, a decentralized system consists of a group of interchangeable machines that can perform as a full system or `distributed database`. Additional machines strengthen the overall system by adding additional redundancy and computational power.
```

**Review by ipopescu — APPROVED** [review]
Nicely done. I left just a couple of nits while reading.

**Review by darthsiroftardis — APPROVED** [review]

**Pull request merged**